crates/autopilot/src/run_loop.rs

```diff
@@ -682,14 +683,14 @@ impl RunLoop {
 
             match self
                 .persistence
-                .find_tx_hash_by_auction_id(auction_id)
+                .find_settlement_transactions(auction_id)
                 .await
             {
-                Ok(Some(hash)) => return Ok(hash),
+                Ok(hashes) if hashes.is_empty() => {}
+                Ok(hashes) => return Ok(hashes.into_iter().next().expect("non-empty")),
```

Comment:
How about avoid using code that panics?
```suggestion
                Ok(hashes) => {
                    if let Some(hash) = hashes.into_iter().next() {
                        return Ok(hash);
                    }
                }
```

Reply:
Why not, whatever makes us sleep better at night :)